mungegithub/mungers/check-labels.go

```diff
@@ -0,0 +1,140 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mungers
+
+import (
+	yamlhelper "github.com/ghodss/yaml"
+	githubhelper "k8s.io/contrib/mungegithub/github"
+
+	"fmt"
+	"github.com/golang/glog"
+	"github.com/google/go-github/github"
+	"github.com/spf13/cobra"
+	"io"
+	"k8s.io/contrib/mungegithub/features"
+	"k8s.io/kubernetes/pkg/util/sets"
+	"k8s.io/kubernetes/pkg/util/yaml"
+	"os"
+	"strings"
+)
+
+type labelAccessor interface {
+	AddLabel(label *github.Label) error
+	GetLabels() ([]*github.Label, error)
+	GetFileContents(file string) (string, string, error)
+}
+
+// CheckLabelsMunger will check that the labels specified in the labels yaml file
+// are created.
+type CheckLabelsMunger struct {
+	labelFileName string
+	lastSha       string
+	labelAccessor labelAccessor
+}
+
+func init() {
+	RegisterMungerOrDie(&CheckLabelsMunger{})
+}
+
+// Name is the name usable in --pr-mungers
+func (c *CheckLabelsMunger) Name() string { return "check-labels" }
+
+// RequiredFeatures is a slice of 'features' that must be provided.
+func (c *CheckLabelsMunger) RequiredFeatures() []string { return []string{} }
+
+// Initialize will initialize the munger.
+func (c *CheckLabelsMunger) Initialize(config *githubhelper.Config, features *features.Features) error {
+	file := c.labelFileName
+	if len(file) == 0 {
+		glog.Fatalf("No --label-file= supplied, cannot check labels")
+	}
+	c.labelAccessor = config
+	return nil
+}
+
+// EachLoop is called at the start of every munge loop
+func (c *CheckLabelsMunger) EachLoop() error {
+	fileContents, sha, err := c.labelAccessor.GetFileContents(c.labelFileName)
```

Comment:
@eparis Since we are making mungegithub generic in order to work with other repositories as well, would it be fair to assume that all of them will have the same fetching of the master git tree? I designed it to use the API because then it is independent of whether the blunderbuss assigner is being run. However, if we will be running blunderbuss against other repositories in the same way, I'll go ahead and make the change to read the local file from the pull for blunderbuss.